src/AvionicsUplink.cpp

```diff
@@ -39,6 +53,108 @@ void Avionics::parseCommand(int16_t len) {
   }
 }
 
+bool compareTime(PlannedCommand command1, PlannedCommand command2) {
+  if(command1.TIMESTAMP==command2.TIMESTAMP) return (command1.COMMAND_INDEX<command2.COMMAND_INDEX);
+  return (command1.TIMESTAMP<command2.TIMESTAMP);
+}
+
+/*
+ * Function: parseCommandNew
+ * -------------------
+ * This function parses planned commands received from the RockBLOCK and places them in an array.
+ */
+ void Avionics::parseCommandNew() {
+   for(uint8_t i=0; i<PLANNED_COMMANDS_SIZE; i++) {
+     PLANNED_COMMANDS[i].COMMAND_INDEX = -1;
+     PLANNED_COMMANDS[i].TIMESTAMP = UINT32_MAX;
+     PLANNED_COMMANDS[i].COMMAND_VALUE = -1;
+   } // erases all planned commands
+   memset(shouldInterpolate, 0, sizeof(shouldInterpolate)); // resets shouldInterpolate
+   memset(hasPlans, 0, sizeof(hasPlans)); // resets hasPlans
+   uint32_t plannedIndex = 0;
+   char *oneComm;
+   oneComm = strtok(COMMS_BUFFER," ");
+   while(oneComm != NULL) {
```

Comment:
This parsing _looks_ correct to me, but is complex enough that we should test it thoroughly